diff --git a/include/fmt/chrono.h b/include/fmt/chrono.h
index 9fbeeed6..6d6e7031 100644
--- a/include/fmt/chrono.h
+++ b/include/fmt/chrono.h
@@ -518,7 +518,7 @@ inline auto gmtime(std::time_t time) -> std::tm {
 
     inline auto fallback(int res) -> bool { return res == 0; }
 
-#if !FMT_MSC_VERSION
+#if !FMT_MSC_VERSION || NXDK
     inline auto fallback(detail::null<>) -> bool {
       std::tm* tm = std::gmtime(&time_);
       if (tm) tm_ = *tm;
diff --git a/include/fmt/ostream.h b/include/fmt/ostream.h
index bf2371b7..89643d63 100644
--- a/include/fmt/ostream.h
+++ b/include/fmt/ostream.h
@@ -12,6 +12,7 @@
 #  include <fstream>  // std::filebuf
 #endif
 
+#ifndef NXDK
 #ifdef _WIN32
 #  ifdef __GLIBCXX__
 #    include <ext/stdio_filebuf.h>
@@ -19,6 +20,7 @@
 #  endif
 #  include <io.h>
 #endif
+#endif
 
 #include "chrono.h"  // formatbuf
 
@@ -127,6 +129,7 @@ inline void vprint(std::ostream& os, string_view fmt, format_args args) {
   else if (auto* fbuf = dynamic_cast<__gnu_cxx::stdio_filebuf<char>*>(rdbuf))
     f = fbuf->file();
 #endif
+#ifndef NXDK
 #ifdef _WIN32
   if (f) {
     int fd = _fileno(f);
@@ -135,6 +138,7 @@ inline void vprint(std::ostream& os, string_view fmt, format_args args) {
       if (detail::write_console(fd, {buffer.data(), buffer.size()})) return;
     }
   }
+#endif
 #endif
   detail::ignore_unused(f);
   detail::write_buffer(os, buffer);
diff --git a/include/fmt/xchar.h b/include/fmt/xchar.h
index 9334b87f..a523ad36 100644
--- a/include/fmt/xchar.h
+++ b/include/fmt/xchar.h
@@ -287,6 +287,7 @@ inline auto formatted_size(const S& fmt, T&&... args) -> size_t {
   return buf.count();
 }
 
+#ifndef NXDK
 inline void vprint(std::FILE* f, wstring_view fmt, wformat_args args) {
   auto buf = wmemory_buffer();
   detail::vformat_to(buf, fmt, args);
@@ -298,6 +299,7 @@ inline void vprint(std::FILE* f, wstring_view fmt, wformat_args args) {
 inline void vprint(wstring_view fmt, wformat_args args) {
   vprint(stdout, fmt, args);
 }
+#endif
 
 template <typename... T>
 void print(std::FILE* f, wformat_string<T...> fmt, T&&... args) {
